runtime/src/qio/qio_formatted.c

```diff
@@ -395,7 +457,12 @@ qioerr qio_channel_read_string(const int threadsafe, const int byteorder, const
 
   err = qio_channel_mark(false, ch);
   if( err ) goto unlock;
-
+  if(str_style==QIO_BINARY_STRING_STYLE_TOEOF)
+    {
+       //printf("Inside the new if condition");
+       err=read_entire_file(ch,out,len_out,maxlen);
+       return err;
```

Comment:
Returning here will cause the channel never to be unlocked.
Consider one of these two options:
 1. Use `goto unlock` here and make sure the following logic makes sense in this setting or
 2. Move your check for `QIO_BINARY_STRING_STYLE_TOEOF` and the call to `read_entire_file` before the `if( threadsafe )` block above that does the locking, and then handle locking and unlocking inside of `read_entire_file`.

I think I have a slight preference for 2.

Either way, your new function should be called before the `qio_channel_mark` call.

Reply:
Changes made as requested .